inference-engine/src/transformations/src/transformations/common_optimizations/stride_optimization.cpp

```diff
@@ -0,0 +1,198 @@
+    second.replace_source_output(pool);
+}
+
+static void handle_not_equal_stride_props(const std::shared_ptr<ngraph::Node>& node, std::vector<ngraph::Input<ngraph::Node>>&& next_ops) {
+    for (auto& op : next_ops) {
+        if (!has_strides_prop(op))
+            continue;
+        auto strides = get_strides_prop(op);
+        bool are_strides_ones = std::all_of(strides.begin(), strides.end(),
+                                            [] (size_t s) -> bool { return s == 1; });
+        if (!are_strides_ones) {
+            auto conv = dynamic_cast<ngraph::opset7::Convolution*>(op.get_node());
+            if (conv) {
+                conv->set_strides(strides);
+            } else {
+                insert_pooling(node, op, strides);
+            }
+        }
+    }
+}
+
+ngraph::pass::ConvStridePropagation::ConvStridePropagation() {
+    MATCHER_SCOPE(ConvStridePropagation);
+    auto data = pattern::any_input();
+    auto weights = pattern::wrap_type<opset7::Constant>();
+    auto conv_pattern = ngraph::pattern::wrap_type<opset7::Convolution>({data, weights});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto conv = std::dynamic_pointer_cast<opset7::Convolution>(m.get_match_root());
+        if (!conv)
+            return false;
+
+        auto conv_strides = conv->get_strides();
+        Strides strides_ones(conv_strides.size(), 1);
+        auto next_ops = get_node_target_inputs(conv);
+        bool all_ops_are_valid;
+        std::vector<Strides> strides_vec;
+        std::tie(strides_vec, all_ops_are_valid) = check_next_ops(next_ops);
+
+        if (!all_ops_are_valid) {
+            handle_not_equal_stride_props(conv, std::move(next_ops));
+        } else if (strides_vec[0].size() > 0) {
+            std::transform(conv_strides.begin(), conv_strides.end(), strides_vec[0].begin(), conv_strides.begin(),
+                    [] (size_t s1, size_t s2) -> size_t { return s1 * s2; });
+        }
+
+        if (can_propagate_conv_stride(conv)) {
+            conv->set_strides(strides_ones);
+            insert_strides_prop(conv->input(0), conv_strides);
+        } else {
+            conv->set_strides(conv_strides);
+        }
+
+        return true;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(conv_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SupportedNodesStridePropagation, "SupportedNodesStridePropagation", 0);
+
+ngraph::pass::SupportedNodesStridePropagation::SupportedNodesStridePropagation() {
+    MATCHER_SCOPE(SupportedNodesStridePropagation);
+    auto root = pattern::wrap_type<opset7::Add,
+                                   opset7::Relu,
+                                   opset7::PRelu,
+                                   opset7::Maximum,
+                                   opset7::Multiply>();
```

Comment:
You need to divide this pass to two:
1. Handles `util::UnaryElementwiseArithmetic` and other ops with single input
2. Handles `util::BinaryElementwiseArithmetic` operations. And for this operation we need to propagate Stride attribute to the first and second input. Also we need to handle Constant operation and in case if stride != 1 for some dimension we need to insert StridedSlice operation.

Reply:
I tried to split this pass, but then I realize I can handle Constant the same way I handle unsupported operations.